diff --git a/lib/minify/min/utils.php b/lib/minify/min/utils.php
--- a/lib/minify/min/utils.php
+++ b/lib/minify/min/utils.php
@@ -75,16 +75,17 @@ function Minify_groupsMtime($groups)
  * @param string $group a key from groupsConfig.php
  * @return Minify_Build
  * @private
  */
 function _Minify_getBuild($group)
 {
     static $builds = array();
     static $gc = false;
+    global $min_groupConfigPath;
     if (false === $gc) {
-        $gc = (require dirname(__FILE__) . '/groupsConfig.php');
+        $gc = (require $min_groupConfigPath);
     }
     if (! isset($builds[$group])) {
         $builds[$group] = new Minify_Build($gc[$group]);
     }
     return $builds[$group];
 }
