diff --git a/lib/php5redis/Redis.php b/lib/php5redis/Redis.php
--- a/lib/php5redis/Redis.php
+++ b/lib/php5redis/Redis.php
@@ -12,17 +12,21 @@ class Redis {
 	private $host;
 	private $_sock;
 	public $debug=false;
 	
 	function __construct($host='localhost', $port = 6379) {
 		$this->host = $host;
 		$this->port = $port;
 	}
-	
+
+	function __destruct() {
+		$this->disconnect();
+	}
+
 	private function connect() {
 		if ($this->_sock)
 			return;
 		if ($sock = fsockopen ( $this->host, $this->port, $errno, $errstr )) {
 			$this->_sock = $sock;
 			$this->debug('Connected');
 			return;
 		}
@@ -96,17 +100,17 @@ class Redis {
 	function pipeline_responses(){
 		$response = array();
 		for ($i=0;$i<$this->pipeline_commands;$i++){
 			$response[] = $this->cmdResponse();
 		}
 		$this->pipeline = false;
 		return $response;
 	}
-	private function cmd($command) {
+	private function cmd($command, $readResp = true) {
 		$this->debug('Command: '.(is_array($command)?join(', ',$command):$command));
 		$this->connect ();
 		
 		if (is_array($command)){
 			// Use unified command format
 			
 			$s = '*'.count($command)."\r\n";
 			foreach ($command as $m){
@@ -122,19 +126,21 @@ class Redis {
 			if ($i == 0)
 				break;
 			$s = substr ( $s, $i );
 		}
 		if ($this->pipeline){
 			$this->pipeline_commands++;
 			return null;
 		}
-		else{
+		elseif ($readResp) {
 			return $this->cmdResponse ();
 		}
+		else
+			return '';
 	}
 	function disconnect() {
 		if ($this->_sock)
 			@fclose ( $this->_sock );
 		$this->_sock = null;
 	}
 	
 	////////////////////////////////
@@ -144,17 +150,18 @@ class Redis {
 	/**
 	 * close the connection
 	 * 
 	 * Ask the server to silently close the connection. 
 	 * 
 	 * @return void The connection is closed as soon as the QUIT command is received. 
 	 */
 	function quit() {
-		return $this->cmd ( 'QUIT' );
+		$this->cmd ( 'QUIT', false );
+		$this->_sock = null;
 	}
 	
 	/**
 	 * simple password authentication if enabled
 	 * 
 	 * Request for authentication in a password protected Redis server. A Redis server 
 	 * can be instructed to require a password before to allow clients to issue commands. 
 	 * This is done using the requirepass directive in the Redis configuration file. 
