--- class.phpmailer.php.original	2008-11-29 21:39:21.000000000 +0100
+++ class.phpmailer.php	2008-12-23 22:11:41.000000000 +0100
@@ -249,23 +249,23 @@
   public $LE              = "\r\n";
 
   /////////////////////////////////////////////////
-  // PROPERTIES, PRIVATE
+  // PROPERTIES, PROTECTED
   /////////////////////////////////////////////////
 
-  private $smtp            = NULL;
-  private $to              = array();
-  private $cc              = array();
-  private $bcc             = array();
-  private $ReplyTo         = array();
-  private $attachment      = array();
-  private $CustomHeader    = array();
-  private $message_type    = '';
-  private $boundary        = array();
-  private $language        = array();
-  private $error_count     = 0;
-  private $sign_cert_file  = "";
-  private $sign_key_file   = "";
-  private $sign_key_pass   = "";
+  protected $smtp            = NULL;
+  protected $to              = array();
+  protected $cc              = array();
+  protected $bcc             = array();
+  protected $ReplyTo         = array();
+  protected $attachment      = array();
+  protected $CustomHeader    = array();
+  protected $message_type    = '';
+  protected $boundary        = array();
+  protected $language        = array();
+  protected $error_count     = 0;
+  protected $sign_cert_file  = "";
+  protected $sign_key_file   = "";
+  protected $sign_key_pass   = "";
 
   /////////////////////////////////////////////////
   // METHODS, VARIABLES
@@ -527,9 +527,9 @@
 
     $smtp_from = ($this->Sender == '') ? $this->From : $this->Sender;
     if(!$this->smtp->Mail($smtp_from)) {
+      $this->smtp->Reset();
       $error = $this->Lang('from_failed') . $smtp_from;
       $this->SetError($error);
-      $this->smtp->Reset();
       return false;
     }
 
@@ -557,15 +557,15 @@
         }
         $error .= $bad_rcpt[$i];
       }
+      $this->smtp->Reset();
       $error = $this->Lang('recipients_failed') . $error;
       $this->SetError($error);
-      $this->smtp->Reset();
       return false;
     }
 
     if(!$this->smtp->Data($header . $body)) {
-      $this->SetError($this->Lang('data_not_accepted'));
       $this->smtp->Reset();
+      $this->SetError($this->Lang('data_not_accepted'));
       return false;
     }
     if($this->SMTPKeepAlive == true) {
@@ -614,8 +614,8 @@
 
         if($tls) {
           if(!$this->smtp->StartTLS()) {
-            $this->SetError($this->Lang("tls"));
             $this->smtp->Reset();
+            $this->SetError($this->Lang("tls"));
             $connection = false;
           }
 
@@ -626,8 +626,8 @@
         $connection = true;
         if($this->SMTPAuth) {
           if(!$this->smtp->Authenticate($this->Username, $this->Password)) {
-            $this->SetError($this->Lang('authenticate'));
             $this->smtp->Reset();
+            $this->SetError($this->Lang('authenticate'));
             $connection = false;
           }
         }
@@ -1231,11 +1231,11 @@
       $this->SetError($this->Lang('file_open') . $path);
       return '';
     }
-    if (function_exists('get_magic_quotes')) {
+    if (!function_exists('get_magic_quotes')) {
         function get_magic_quotes() {
             return false;
         }
-}
+    }
     if (PHP_VERSION < 6) {
       $magic_quotes = get_magic_quotes_runtime();
       set_magic_quotes_runtime(0);
@@ -1622,7 +1622,7 @@
    * @access private
    * @return void
    */
-  private function SetError($msg) {
+  protected function SetError($msg) {
     $this->error_count++;
     $this->ErrorInfo = $msg;
   }
@@ -1664,7 +1664,7 @@
    * @access private
    * @return string
    */
-  private function Lang($key) {
+  protected function Lang($key) {
     if(count($this->language) < 1) {
       $this->SetLanguage('en'); // set the default language
     }
