--- class.phpmailer.php.original	2010-12-11 10:53:14.000000000 +0100
+++ class.phpmailer.php	2011-05-01 18:34:13.000000000 +0200
@@ -306,21 +306,21 @@
   // PROPERTIES, PRIVATE AND PROTECTED
   /////////////////////////////////////////////////
 
-  private   $smtp           = NULL;
-  private   $to             = array();
-  private   $cc             = array();
-  private   $bcc            = array();
-  private   $ReplyTo        = array();
-  private   $all_recipients = array();
-  private   $attachment     = array();
-  private   $CustomHeader   = array();
-  private   $message_type   = '';
-  private   $boundary       = array();
+  protected $smtp           = NULL;
+  protected $to             = array();
+  protected $cc             = array();
+  protected $bcc            = array();
+  protected $ReplyTo        = array();
+  protected $all_recipients = array();
+  protected $attachment     = array();
+  protected $CustomHeader   = array();
+  protected $message_type   = '';
+  protected $boundary       = array();
   protected $language       = array();
-  private   $error_count    = 0;
-  private   $sign_cert_file = "";
-  private   $sign_key_file  = "";
-  private   $sign_key_pass  = "";
+  protected $error_count    = 0;
+  protected $sign_cert_file = "";
+  protected $sign_key_file  = "";
+  protected $sign_key_pass  = "";
   private   $exceptions     = false;
 
   /////////////////////////////////////////////////
@@ -449,7 +449,7 @@
    * @return boolean true on success, false if address already used or invalid in some way
    * @access private
    */
-  private function AddAnAddress($kind, $address, $name = '') {
+  protected function AddAnAddress($kind, $address, $name = '') {
     if (!preg_match('/^(to|cc|bcc|ReplyTo)$/', $kind)) {
       echo 'Invalid recipient array: ' . kind;
       return false;
@@ -569,14 +569,11 @@
       }
 
       // Choose the mailer and send through it
-      switch($this->Mailer) {
-        case 'sendmail':
-          return $this->SendmailSend($header, $body);
-        case 'smtp':
-          return $this->SmtpSend($header, $body);
-        default:
-          return $this->MailSend($header, $body);
-      }
+      $method = $this->Mailer.'Send';
+      if (method_exists($this, $method))
+        return $this->{$method}($header, $body);
+      else
+        return $this->MailSend($header, $body);
 
     } catch (phpmailerException $e) {
       $this->SetError($e->getMessage());
@@ -668,7 +665,7 @@
     } else {
       if ($this->SingleTo === true && count($toArr) > 1) {
         foreach ($toArr as $key => $val) {
-          $rt = @mail($val, $this->EncodeHeader($this->SecureHeader($this->Subject)), $body, $header, $params);
+          $rt = @mail($val, $this->EncodeHeader($this->SecureHeader($this->Subject)), $body, $header);
           // implement call back function if it exists
           $isSent = ($rt == 1) ? 1 : 0;
           $this->doCallback($isSent,$val,$this->cc,$this->bcc,$this->Subject,$body);
@@ -1468,11 +1465,11 @@
       }
       if (PHP_VERSION < 6) {
         $magic_quotes = get_magic_quotes_runtime();
-        set_magic_quotes_runtime(0);
+        ini_set('magic_quotes_runtime', 0);
       }
       $file_buffer  = file_get_contents($path);
       $file_buffer  = $this->EncodeString($file_buffer, $encoding);
-      if (PHP_VERSION < 6) { set_magic_quotes_runtime($magic_quotes); }
+      if (PHP_VERSION < 6) { ini_set('magic_quotes_runtime', $magic_quotes); }
       return $file_buffer;
     } catch (Exception $e) {
       $this->SetError($e->getMessage());
@@ -1953,7 +1950,7 @@
    * @access private
    * @return string
    */
-  private function Lang($key) {
+  protected function Lang($key) {
     if(count($this->language) < 1) {
       $this->SetLanguage('en'); // set the default language
     }
@@ -2317,4 +2314,3 @@
     return $errorMsg;
   }
 }
-?>
\ No newline at end of file
